### DISCUSSION TITLE
Instructions for serial plugin in Nornir 3.1.0?

### DISCUSSION THREAD

**Discussion by lucasmarcel** [nornir-automation/nornir#675] [Q&A]
I set
```
runners:
    plugin: serial
```
in config.yaml. I then inserted ```import ipdb; ipdb.settrace()``` in my custom task to troubleshoot, but the script doesn't stop at the first invocation of the custom task for the first host as expected. Instead, it seems to loop through all the hosts and calling my custom task in rapid succession.
Is there another configuration to change to have nornir process the hosts one by one when troubleshooting?
I tried:
```
runners:
    plugin: threaded
    options:
        num_workers: 1
```
but it didn't work either.

My code in a nutshell:
```
nr = InitNornir(config_file="config.yaml")

def mytask(task):
    import ipdb; ipdb.set_trace()
    print("In mytask")
    
nr.run(task=mytask)
```

**Comment by dbarrosop** [comment]
I just tested your example and works fine. Are you sure you are passing the correct `config.yaml`? What is the output of `print(nr.runner)`?


 You can also try this:

``` python
from nornir.init_nornir import InitNornir
from nornir.plugins.runners import SerialRunner


nr = InitNornir(config_file="config.yaml")


def mytask(task):
    import ipdb; ipdb.set_trace()
    print("In mytask")
    


nr.with_runner(SerialRunner()).run(task=mytask)
```

**Reply by lucasmarcel** [reply]
Thanks, I found out what was wrong.
In config.yaml, the expected key name is "runner" apparently, not "runner**s**".
I got the plural form from the doc [https://nornir.readthedocs.io/en/latest/upgrading/2_to_3.html#num-workers](url).
I would probably be a good idea to update it.

Printing the number of workers (```print(nr.runner.num_workers)```) showed me that even with a threaded runner, the number of threads I specified in config.yaml was ignored (default seems to be 20). It made me realize that this portion of config.yaml was not being processed.

Importing SerialRunner in the script itself worked also, thanks for showing me it was an option.

And thanks again for maintaining this project.
